### ISSUE TITLE
[bug] wallet_addEthereumChain or useSwitchChain on Rainbow Wallet not working

### ISSUE THREAD

**Issue by kingnight153** [rainbow-me/rainbowkit#1791]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### RainbowKit Version

2.0.0

### wagmi Version

2.5.6

### Current Behavior

I already add a custom chain to `getDefaultConfig` 
1. When I used `switchChain` from `useSwitchChain` wagmi for Rainbow wallet to switch to this chain, error occured like this:

<img width="325" alt="image" src="https://github.com/rainbow-me/rainbowkit/assets/68669216/11b85188-2dfa-4113-8628-660cd417ef9e">

When I used `switchChain` for other wallets like MetaMask, it worked, but rainbow didn't

3. When I tried using `wallet_addEthereumChain` to add this chain to Rainbow wallet, though I approved the addition of this chain, the error occured. I checked Rainbow wallet networks, this custom chain is already added but error showed: 
<img width="355" alt="image" src="https://github.com/rainbow-me/rainbowkit/assets/68669216/566f191d-bfac-4711-9732-5a6ba3f0df56">

<img width="292" alt="image" src="https://github.com/rainbow-me/rainbowkit/assets/68669216/b851a12f-e33c-4710-a07b-3e6719f06754">

### Expected Behavior

When I used `switchChain` from `useSwitchChain` wagmi for Rainbow wallet to switch to a custom chain, it should add the custom chain to RainBow wallet and switch to it

If I use `wallet_addEthereumChain`, Rainbow wallet should add the chain and not show any error

**Comment by KosmosKey** [comment]
@kingnight153 Could you please tell what chain you're switching to ? 

Also would be super helpful if you can show your RainbowKit and Wagmi configuration setup 🙏

**Comment by kingnight153** [comment]
Here is the chain info: https://chainlist.org/?search=carbon
<img width="551" alt="image" src="https://github.com/rainbow-me/rainbowkit/assets/68669216/85d6b3bc-69ef-4b2a-afa4-9897cfe01262">

**Comment by KosmosKey** [comment]
Yeah so that's an unsupported network from rainbow wallet. We work on improving the custom network so hopefully should be fixed soon. What you can do now is add the network manually and try to switch to the chain.

![image](https://github.com/rainbow-me/rainbowkit/assets/53529533/29f2c312-44b6-4fe9-978c-c9d58fa0dcea)

**Comment by DanielSinclair** [comment]
@kingnight153 Thanks for reporting. Bumping this to our browser-extension repo to further investigate
